src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/noenumtrailingcomma/InputNoEnumTrailingComma.java

```diff
@@ -0,0 +1,87 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.noenumtrailingcomma;
+
+public class InputNoEnumTrailingComma {
+
+    enum Foo1 {
+        FOO,
+        BAR; //OK
+    }
+
+    enum Foo2 {
+        FOO,
+        BAR //OK
+    }
+
+    enum Foo3 {
+        FOO,
+        BAR, //violation
+    }
+
+    enum Foo4 {
+        FOO,
+        BAR, //violation
+        ;
+    }
+
+    enum Foo5 {
+        FOO,
+        BAR,; //violation
+    }
+
+    enum Foo6 { FOO, BAR,; } //violation
+
+    enum Foo7 { FOO, BAR, } //violation
+
+    enum Foo8 {
+        FOO,
+        BAR //OK
+        ;
+    }
+
+    enum Foo9 { FOO, BAR; } //OK
+
+    enum Foo10 { FOO, BAR } //OK
+
+    enum Foo11 {} //OK
+
+    enum Foo12 { FOO, } //violation
+
+    enum Foo13 { FOO } // OK
+
+    enum Foo14 {
+        FOO, //violation
+    }
+
+    enum Foo15 {
+        FOO //OK
+    }
+
+    enum Foo16 {
+        //OK
+    }
+
+    enum Foo17 { FOO; } //OK
+
+    enum Foo18 {
+        FOO; //OK
+    }
+
+    enum Foo19 {
+        FOO
+        , //violation
+    }
+
+    enum Foo20 {
+        FOO
+        ; //OK
+    }
+
+    public enum Foo21
+    {
+        FIRST_CONSTANT,
+        SECOND_CONSTANT;
+
+        public void someMethod() {
+        }
+    } //OK
+}
```

Comment:
Please add a test case(s) for complex enums, for example
```
    public enum Complex {
        ONE_ONE(1,1),
        ONE_TWO(1,2);

        int major, minor;

        Complex(int major, int minor) {
            this.major = major;
            this.minor = minor;
        }
    }
```

Discovered [here](https://sd1998.github.io/checkstyle-regression/Fix-6972/elasticsearch/index.html#A2). There is additional comma in the line
`private final Number minValue, maxValue;`

Reply:
Also some cases like overridden methods in enum definition, empty parentheses after enum constants are missing.
@sd1998 you can check more test inputs for other checks for enums, e.g. https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicoloninenumeration/InputUnnecessarySemicolonInEnumeration.java

Reply:
Donee